components/Chips/src/MDCChipField.h

```diff
@@ -0,0 +1,242 @@
+ by VoiceOver users on all versions of iOS.
+ */
+
+/** The minimum text field width of the text field contained within the MDCChipField. */
+extern const CGFloat kMDCChipFieldDefaultMinTextFieldWidth;
+
+/** The default content edge insets for the MDCChipField. */
+extern const UIEdgeInsets kMDCChipFieldDefaultContentEdgeInsets;
+
+@protocol MDCChipFieldDelegate;
+
+/** Defines when text field input is changed into a chip. */
+typedef NS_OPTIONS(NSUInteger, MDCChipFieldDelimiter) {
+  MDCChipFieldDelimiterNone = 0,
+  MDCChipFieldDelimiterReturn = 1 << 0,
+  MDCChipFieldDelimiterSpace = 1 << 1,
+  MDCChipFieldDelimiterDidEndEditing = 1 << 2,
+  MDCChipFieldDelimiterDefault = (MDCChipFieldDelimiterReturn | MDCChipFieldDelimiterDidEndEditing),
+  MDCChipFieldDelimiterAll = 0xFFFFFFFF
+};
+
+@interface MDCChipField : UIView
+
+/**
+ The text field used to enter new chips.
+
+ Do not set the delegate or positioningDelegate.
+
+ If you set a custom font, make sure to also set the custom font on textField.placeholderLabel and
+ on your MDCChipView instances.
+ */
+@property(nonatomic, nonnull, readonly) MDCTextField *textField;
+
+/**
+ The fixed height of all chip views.
+
+ Default is 32.
+ */
+@property(nonatomic, assign) CGFloat chipHeight;
+
+/**
+ Attribute to determine whether to show the placeholder text (if it exists) when chips are
+ present.
+
+ Default is YES.
+ */
+@property(nonatomic, assign) BOOL showPlaceholderWithChips;
+
+/**
+ The delimiter used to create chips in the text field. Uses default value
+ MDCChipFieldDelimiterDefault if no delimiter is set.
+ */
+@property(nonatomic, assign) MDCChipFieldDelimiter delimiter;
+
+/**
+ The minimum width of the text field.
+
+ Default is |kMDCChipFieldDefaultMinTextFieldWidth|.
+ */
+@property(nonatomic, assign) CGFloat minTextFieldWidth;
+
+/**
+ The chips that are visible in the input area.
+ */
+@property(nonatomic, nonnull, strong) NSArray<MDCChipView *> *chips;
+
+/**
+ Delegate to receive updates to the chip field view. Implement
+ |chipFieldHeightDidChange| to receive updates when the height of the chip field changes.
+ */
+@property(nonatomic, nullable, weak) id<MDCChipFieldDelegate> delegate;
+
+/**
+ The inset or outset margins for the rectangle surrounding all of the chip field's content.
+ Default is |kMDCChipFieldDefaultContentEdgeInsets|.
+ */
+@property(nonatomic, assign) UIEdgeInsets contentEdgeInsets;
+
+/**
+ Adds a chip to the chip field.
+
+ @param chip The chip to add to the field.
+
+ Note: Implementing |chipField:shouldAddChip| only affects whether user interface input entered into
+ the text field is changed into chips and will not affect use of this method.
+ */
+- (void)addChip:(nonnull MDCChipView *)chip;
+
+/**
+ Removes a chip from the chip field.
+
+ @param chip The chip to remove from the field.
+ */
+- (void)removeChip:(nonnull MDCChipView *)chip;
+
+/** Removes all selected chips from the chip field. */
+- (void)removeSelectedChips;
```

Comment:
There isn't. Currently selected is only used to delete the last token while pressing backspace. It could be argued we should make private the selection related API, but removeSelectedChips is exposed if clients want to customize their backspace behavior.

Reply:
Got it!